### DISCUSSION TITLE
Consider making HTMLKit extensible by default

### DISCUSSION THREAD

**Discussion by tonyarnold** [vapor-community/HTMLKit#83] [General]
I am gradually adding things that I personally need via PRs, however I've noticed that core protocols like `ContentNode` are `internal`, which means that I can't meaningfully add support for new elements. 

I feel like the scope of HTML is too broad to be "locking away the keys" for important protocols like this one, and I'm happy to submit PRs for that, too, but it feels like something worth discussing first?

**Comment by mattesmohr** [comment]
Thanks for giving me the chance to explain myself, but I am not sure if I am understand what you up to. Can you extend your thought?

What do you mean with keys? Do you mean the properties like `key`, `content`, `attributes`.  I don't think there is a reason to make them public since you have the tools to modify them. 

What elements do you have in mind?

**Reply by tonyarnold** [reply]
Sorry, the "keys" comment was a (poor) metaphor. 

I was trying to add support for the `<g>` element within `<svg>` scopes, but couldn't find a way to add or extend new elements. 

This lead me to thinking "this won't be the last time something is missing", but I couldn't find a way to extend HTMLKit to support the new element without going through the process of actually adding it to the project properly via a PR.

My point is that this is a workflow that the project should consider supporting. There's always going to be some corner case HTML element that is missing or not-quite-right, and ideally that'd be fixed here (upstream), but people don't always have time for that.

**Reply by mattesmohr** [reply]
I see your point. Let me think about it, kay?
